### PR TITLE
Update CNI popularity table for 2.8.1

### PR DESCRIPTION
**Pull request by btat**
Related to #1069 

## Description

Updates the table with the current counts as of 2.8.1 release.

Additionally, I've moved the table into a partial under a top-level shared-files directory since the table is the same for all versions. This allows us to only maintain one copy of the content.

### CODE REVIEW

**Review by sunilarjun — APPROVED** [review]
Smoke tested and all tables loaded successfully across versions.

**Comment by martyav** [comment]
Test deployment failed with heap allocation error

File: shared-files/_cni-popularity.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<!-- releaseTask -->
+The following table summarizes different GitHub metrics to give you an idea of each project's popularity and activity. This data was collected in January 2024.
```
Comment (martyav) [inline shared-files/_cni-popularity.md:2]:
```suggestion
The following table summarizes different GitHub metrics to give you an idea of each project's popularity and activity levels. This data was collected in January 2024.
```

**Review by martyav — COMMENTED** [review]
Minor suggestion for style

**Comment by btat** [comment]
> Test deployment failed with heap allocation error

I've reran the build a couple times it and it's failing each time because of heap allocation, but the other PRs I submitted in the same time period built without issues. I want to see if different PRs that might get pushed today have the same issue before bumping the allocation size again.

**Review by martyav — APPROVED** [review]

**Comment by btat** [comment]
This is now waiting on #1076.

**Pull request merged**